### ISSUE TITLE
Error NoneType not iterable on EBS Snapshot attributes

### ISSUE THREAD

**Issue by MAPontes** [Netflix/security_monkey#960]
Hi all,

First of all, awesome tool :)

## Description of issue:

So I installed it in the new config (UI separated from Scheduler, separated from Workers all talking via Redis). After inserting the first account I noticed that after a while it stopped pulling new changes from the account, checking the logs I saw the below:
```
2018-02-23 10:55:00,597 **ERROR**: [X] Task Account Scheduler Exception (Infra/ebssnapshot): 'NoneType' object is not iterable [in /usr/loc
al/src/security_monkey/security_monkey/task_scheduler/tasks.py:45]
2018-02-23 10:55:00,598 ERROR: Traceback (most recent call last):
  File "/usr/local/src/security_monkey/security_monkey/task_scheduler/tasks.py", line 35, in task_account_tech
    reporter_logic(account_name, technology_name)
  File "/usr/local/src/security_monkey/security_monkey/task_scheduler/tasks.py", line 131, in reporter_logic
    monitors = find_changes(account_name, technology_name)
  File "/usr/local/src/security_monkey/security_monkey/task_scheduler/tasks.py", line 213, in find_changes
    audit_changes([account_name], [monitor_name], False, debug)
  File "/usr/local/src/security_monkey/security_monkey/task_scheduler/tasks.py", line 237, in audit_changes
    _audit_changes(account, monitor.auditors, send_report, debug)
  File "/usr/local/src/security_monkey/security_monkey/task_scheduler/tasks.py", line 289, in _audit_changes
    au.audit_objects()
  File "/usr/local/src/security_monkey/security_monkey/auditor.py", line 674, in audit_objects
    method(item)
  File "/usr/local/src/security_monkey/security_monkey/auditors/ebs_snapshot.py", line 49, in check_friendly_access
    for uid in self._get_permissions(item):
  File "/usr/local/src/security_monkey/security_monkey/auditors/ebs_snapshot.py", line 46, in _get_permissions
    return {perm.get(key) for perm in item.config.get('create_volume_permissions') if key in perm}
**TypeError: 'NoneType' object is not iterable**
 [in /usr/local/src/security_monkey/security_monkey/task_scheduler/tasks.py:46]
```
 Checking the code, I believe it tries to get the attributes of the requested item (ebs snapshot in my case), however I used the aws cli to try and get the attributes of a snapshot and noticed that the 'createVolumePermissions' of my ebs snapshots returns and empty list. So I believe that while parsing that into a list the code gets a NoneType object and hence the error.

To work it around I did the following on file "/usr/local/src/security_monkey/security_monkey/auditors/ebs_snapshot.py":

`if (item.config.get('create_volume_permissions') != None):`
`           return {perm.get(key) for perm in item.config.get('create_volume_permissions') if key in perm}`
`else:`
`           return []`

After this I don't see the error in the log anymore, but I'm not a developer so wanted to know if you guys already have this kind of issue in your track.

**Comment by mikegrima** [comment]
What happens if you replace:
```
    return {perm.get(key) for perm in item.config.get('create_volume_permissions') if key in perm}
``` 
with:
```
    return {perm.get(key) for perm in item.config.get('create_volume_permissions', []) if key in perm}
```

**Comment by MAPontes** [comment]
@mikegrima 

I did the change and change report is still running but it already scanned the problematic account and so far so good. That seems to have solved the issue.

Att.,
Marcelo Araújo Pontes

**Comment by mikegrima** [comment]
Neat -- I will make the change now.

**Comment by mikegrima** [comment]
Fix in #961
